californium-core/src/main/java/org/eclipse/californium/core/network/InMemoryMessageExchangeStore.java

```diff
@@ -353,12 +325,8 @@ public boolean registerOutboundResponse(final Exchange exchange) {
 	 * Purges all registered exchanges from this store.
 	 */
 	public void clear() {
-		synchronized (messageIdProvider) {
-			synchronized (exchangesByToken) {
-				exchangesByMID.clear();
-				exchangesByToken.clear();
-			}
-		}
+		exchangesByMID.clear();
```

Comment:
I agree in that invoking `clear()` makes most sense during shutdown. In fact, the method currently is invoked by `stop()` only, so we could simply reduce its visibility to `private`.